pkg/util/openapi/openapi.go

```diff
@@ -110,6 +110,15 @@ func createConfig() *common.Config {
 			}
 			return m
 		},
+
+		GetDefinitionName: func(name string) (string, spec.Extensions) {
```

Comment:
Could you explain to me what this function exactly does?

Reply:
Every type has a full name (e.g `k8s.io/apimachinery/pkg/apis/meta/v1.Patch`) and corresponding schema(OpenAPIDefinitions in `openapi_generated.go`).
This function takes the full name and should return a friendly name. Default implementation in [kube-openapi](https://github.com/kubernetes/kube-openapi/blob/master/pkg/builder/openapi.go#L103) is to return last part after `/`.

So the solution is to override this function so two identically called types from different sources will not collide...(Note: we still need to make sure we don't have 2 types called identically in our API).